crates/ott-balancer/src/balancer.rs

```diff
@@ -171,12 +171,95 @@ impl BalancerLink {
     }
 }
 
-#[derive(Debug, Default)]
+#[derive(Debug)]
 pub struct BalancerContext {
     pub clients: HashMap<ClientId, BalancerClient>,
     pub monoliths: HashMap<MonolithId, BalancerMonolith>,
     pub rooms_to_monoliths: HashMap<RoomName, RoomLocator>,
     pub monoliths_by_region: HashMap<String, Vec<MonolithId>>,
+    pub monolith_selection: Box<dyn MonolithSelection + Send + Sync + 'static>,
+}
+#[derive(Debug, Default)]
+pub struct MonolithRegistry;
+pub trait MonolithSelection: std::fmt::Debug {
+    fn select_monolith(
+        &self,
+        monolith: Vec<&BalancerMonolith>,
+        ctx: &BalancerContext,
+    ) -> anyhow::Result<&BalancerMonolith>;
```

Comment:
The reason the compiler is complaining about lifetimes is because it can't automatically guarantee that the reference you are returning isn't something you fabricated in the function. The lifetime annotations tell the compiler that yes, the parameters and the returned object must live for the same amount of time. The compiler diagnostic gives a hint for what code changes need to be made if you look at the output in your terminal.